src/autoscaler/api/publicapiserver/route_helper.go

```diff
@@ -144,11 +136,11 @@ func paginateResource(resourceList []byte, parameters *url.Values, r *http.Reque
 	return result, nil
 }
 
-func getPageUrl(r *http.Request, parameters *url.Values, targetPageNo int) string {
+func getPageUrl(r *http.Request, queries *url.Values, targetPageNo int) string {
 	pageUrl, _ := url.Parse(r.URL.String())
 
 	pageParams := url.Values{}
-	for key, value := range *parameters {
+	for key, value := range *queries {
```

Comment:
The "queries" seems unnecessary, since it derives from "r".  Also, the queries & parameters here looks confusion.  The "pageNo" is still extracted from "parameter", then why we need to enforce the usage of "query" instead of "parameter" ?